R/utilities-postprocessing.R

```diff
@@ -23,24 +23,45 @@ release2json <- function(release, nameslist){
     names(initialized) <- names
     
     for(i in 1:k){
-        att <- nameslist[[i]] #release[[i]]$result$variable
+        att <- namesList[[i]] #release[[i]]$result$variable
         if(!initialized[[att]]){
             variables[[att]] <- createfields(variables[[att]], release[[i]], att)
         }
         variables[[att]] <- fillfields(variables[[att]], release[[i]])
     }
     
     
-    dataset_metadata<-list()
-    dataset_metadata$private <- TRUE
+    datasetMetadata<-list()
+    datasetMetadata$private <- TRUE
     
-    releasedMetadata <- list(dataset=dataset_metadata, variables=variables)
+    releasedMetadata <- list(dataset=datasetMetadata, variables=variables)
     result <- jsonlite:::toJSON(releasedMetadata, digits=8)
     
     return(result)
     
 }
 
+#' Function to create JSON file defining differentially private statistics
+
+createJSON <- function() { 
+    
+    statistics <- list(
+        'histogram', 
+        'mean'
+    )
+    
+    statJSON <- function(stat) {
+        func <- match.fun(paste0(stat, '.getJSON'))
```

Comment:
To preserve the behavior before the refactor, you'll need to fix the reference to the function:

```suggestion
func <- match.fun(paste0(stat, 'GetJSON'))
```

That being said, IMO, match.fun is eval's little brother. And string-evals tend to get missed when refactoring. Here's another option:

```suggestion
func <- list(
  histogram=histogramGetJSON,
  mean=meanGetJSON
)[[stat]]
```